### DISCUSSION TITLE
CleanupTime Parameter not working for Adaptive Backups

### DISCUSSION THREAD

**Discussion by bawedy** [olahallengren/sql-server-maintenance-solution#841] [Questions & Answers]
I'm using the adaptive backup feature with the following parameters
- BackupType = Diff
- ChangeBackupType = 'Y'
- ModificationLevel  = 50
- CleanupTime = 180

When the backuptype remains as Diff, then the XP_DeleteFile procedure is called as expected and cleans up backup files as per the CleanupTime parameter; however when the BackupType gets changed to Full, the XP_DeleteFile procedure does not get called and therefore no cleanup happens; this looks as if the change of backuptype from "Diff" to "Fuill" is preventing the cleanup.

**Comment by olahallengren** [comment]
This is by design.

```
      -- Delete old backup files, after backup
      IF ((@CurrentBackupOutput = 0 AND @Verify = 'N')
      OR (@CurrentBackupOutput = 0 AND @Verify = 'Y' AND NOT EXISTS (SELECT * FROM @CurrentBackupSet WHERE VerifyOutput <> 0 OR VerifyOutput IS NULL)))
      AND (@BackupSoftware <> 'DATA_DOMAIN_BOOST' OR @BackupSoftware IS NULL)
      AND @CurrentBackupType = @BackupType
```

The reason for this is that you might have different retention settings for the different backup types. Imagine that you have 168 hours retention for your full backups, but only 48 hours for your differential backups. Doing cleanup after a change of backup type, could then cause your full backups to be deleted after 48 hours.

**Reply by bawedy** [reply]
Thanks Ola
In that case I will comment out that bit in the code then
Thanks for taking the time to look in to that
